desktop/src/gui/preferences_dialog.rs

```diff
@@ -0,0 +1,169 @@
+        let locked_text = text(&self.locale, "preference-locked-by-cli");
+
+        Window::new(text(&self.locale, "preferences-dialog"))
+            .open(&mut keep_open)
+            .anchor(Align2::CENTER_CENTER, egui::Vec2::ZERO)
+            .collapsible(false)
+            .resizable(false)
+            .show(egui_ctx, |ui| {
+                ui.vertical_centered_justified(|ui| {
+                    Grid::new("preferences-dialog-graphics")
+                        .num_columns(2)
+                        .striped(true)
+                        .show(ui, |ui| {
+                            self.show_graphics_preferences(&locked_text, ui);
+                        });
+
+                    if self.restart_required() {
+                        ui.colored_label(
+                            ui.style().visuals.error_fg_color,
+                            "A restart is required to apply the selected changes",
+                        );
+                    }
+
+                    ui.horizontal(|ui| {
+                        ui.with_layout(egui::Layout::right_to_left(egui::Align::Center), |ui| {
+                            if Button::new(text(&self.locale, "save")).ui(ui).clicked() {
+                                self.save();
+                                should_close = true;
+                            }
+                        })
+                    });
+                });
+            });
+
+        keep_open && !should_close
+    }
+
+    fn restart_required(&self) -> bool {
+        self.graphics_backend != self.preferences.graphics_backends()
+            || self.power_preference != self.preferences.graphics_power_preference()
+    }
+
+    fn show_graphics_preferences(&mut self, locked_text: &str, ui: &mut Ui) {
+        ui.label(text(&self.locale, "graphics-backend"));
+        if self.graphics_backend_readonly {
+            ui.label(graphics_backend_name(&self.locale, self.graphics_backend))
+                .on_hover_text(locked_text);
+        } else {
+            ComboBox::from_id_source("graphics-backend")
+                .selected_text(graphics_backend_name(&self.locale, self.graphics_backend))
+                .show_ui(ui, |ui| {
+                    ui.selectable_value(
+                        &mut self.graphics_backend,
+                        GraphicsBackend::Default,
+                        text(&self.locale, "graphics-backend-default"),
+                    );
+                    ui.selectable_value(
+                        &mut self.graphics_backend,
+                        GraphicsBackend::Vulkan,
+                        text(&self.locale, "graphics-backend-vulkan"),
+                    );
+                    ui.selectable_value(
+                        &mut self.graphics_backend,
+                        GraphicsBackend::Metal,
+                        text(&self.locale, "graphics-backend-metal"),
+                    );
+                    ui.selectable_value(
```

Comment:
I think we should try to only offer usable graphicbackends.

Reply:
Made that change!